shakedown/dcos/__init__.py

```diff
@@ -162,3 +260,86 @@ def _gen_url(url_path):
     """
     from six.moves import urllib
     return urllib.parse.urljoin(dcos_url(), url_path)
+
+
+# Initialize some things.
+transportManager = TransportManager()
+
+
+class HostSession:
+    """Context manager that returns an SSH session to run commands.
+    
+    """
+    def __init__(self, host, username, key_path, verbose):
+        self.host = host
+        self.username = username
+        self.key_path = key_path
+        self.verbose = verbose
+        self.exit_code = -1
+        self.output = ''
+        self.session = None
+    
+    def __enter__(self):
+        """Called when using a `with` closure.
+
+        :return: this session manager
+        :rtype: HostSession
+        """
+        self.session = transportManager.get_session(
+                self.host,
+                self.username,
+                self.key_path)
+        return self
+    
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Executed when the context manager is complete (exits
+        the with closure).
+
+        :param exc_type:
```

Comment:
these param vars seem more important than most to document...  they are abbreviated in a way that I can't discern them...   the names should be more clear IMO... and if not documented.

Reply:
Thanks for catching this!